src/app/components/Input/InputRange.tsx

```diff
@@ -22,7 +22,12 @@ export const InputRange = React.forwardRef<HTMLInputElement, Props>(
 		const handleInput = (value: string) => {
 			const fraction = Math.pow(10, magnitude! * -1);
 			const amount = BigNumber.make(value).times(fraction);
-			setValues([amount.toNumber()]);
+
+			if (amount.isGreaterThan(max)) {
+				return setValues([max]);
```

Comment:
It works as expected, however, automatically changing the value may be unexpected behavior for the user, it is best handled with validation. We can try something like:

```
const rangeValues = [Math.min(values[0], 100)];
<Range values={rangeValues} />
```

Reply:
Good trick and the code is cleaner @luciorubeens :ok_hand:

I removed the code with `isGreaterThan()` and added:

```
const handleInput = (value: string) => {
	const fraction = Math.pow(10, magnitude! * -1);
	const amount = BigNumber.make(value).times(fraction);
	setValues([amount.toNumber()]);
};

const rangeValues = [Math.min(values[0], max)];

<InputCurrency value={rangeValues[0]} />

<Range values={rangeValues} />
```